### DISCUSSION TITLE
Is it possible to control text wrapping inside of a text?

### DISCUSSION THREAD

**Discussion by dotlogix** [fabricjs/fabric.js#7644] [Q&A]
Hey there I'm currently searching for a canvas library.
I really like the api of fabric, but there is one important feature which I would need.

Is it possible to wrap a text within a simple shape?
Would also be fine if I can control the line start and end on a per line / pixel basis.
Writing some code to create a circular shape using that should be easy.

**Comment by ShaMan123** [comment]
it's not very clear what you want to achieve
some visuals would do
Would group suffice?
if you're asking about custom objects - then, sure you can build what you want but it's not simple
I plan to write some post about it

**Comment by asturur** [comment]
no we don't have that actually.
There is a width property on the text, while what you are looking for is more around wrapping the lines with the width of a shape at a certain height.

In the textbox class everything is decided at the wrap method, and that could be easily overridden to do what you want to do, but we don't have anything that slices the shape/path in lines and calculate the width of them.

If you can solve that problem you can also easily wrap the text.
